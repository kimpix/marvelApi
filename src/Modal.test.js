import React from 'react';
import Modal from './Modal';


test('Modal render', (props ={}) => {

    const tree = shallow(
      <Modal containerClassName={jest.fn()} backdropClassName={jest.fn()} isOpen={jest.fn()} onClose={jest.fn()}/>
    );

    expect(tree).toMatchSnapshot();
});


test('Modal containerClass content', (props ={}) => {
    const tree = shallow(
      <Modal containerClassName='Two test are better than once' backdropClassName={jest.fn()} isOpen={jest.fn()} onClose={jest.fn()}/>
    );

    const modalTree = tree.props().className;

    expect(modalTree).toEqual('Two test are better than once');
});


test('Modal backdropClassName content', (props ={}) => {
    const tree = shallow(
      <Modal containerClassName={jest.fn()} backdropClassName='Three test are better than once' isOpen={jest.fn()} onClose={jest.fn()}/>
    );

    const modalTree = render(tree.props().children[1]);

    expect(modalTree[0].attribs.class).toEqual('Three test are better than once');
});


