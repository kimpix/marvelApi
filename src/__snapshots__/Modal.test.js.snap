// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render 1`] = `
<div
  className={[MockFunction]}
>
  <div
    style={
      Object {
        "background": "#fff",
        "left": "50%",
        "padding": "25px",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "zIndex": "9999",
      }
    }
  />
  <div
    className={[MockFunction]}
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "left": "0px",
        "position": "absolute",
        "top": "0px",
        "width": "100%",
        "zIndex": "9998",
      }
    }
  />
</div>
`;
